workers/handshake-manager/src/manager/internal_engine.rs

```diff
@@ -62,22 +62,33 @@ impl HandshakeExecutor {
 
         // Match against each other order in the local book
         for order_id in other_orders {
-            // Same order
+            // Orders must not be the same order
             if network_order.id == order_id {
                 continue;
             }
 
-            // Same wallet
+            // Orders must be in matchable wallets
             let other_wallet_id = self
                 .state
                 .get_wallet_for_order(&order_id)
                 .await?
                 .ok_or_else(|| HandshakeManagerError::State(ERR_NO_WALLET.to_string()))?;
+
             if !self.wallets_can_match(wallet.wallet_id, other_wallet_id) {
                 continue;
             }
 
-            // Lookup the witness used for this order
+            // Orders must be in the same matching pool
+            let my_pool = self.state.get_matching_pool_for_order(&network_order.id).await?;
+            let other_pool = self.state.get_matching_pool_for_order(&order_id).await?;
```

Comment:
why not have `get_matching_pool_for_order` return `Result<String, ...>` instead of `Result<Option<...>>`? We already default to the global matching pool semantically and it just complicates interfaces to express this as an option when there really isn't one.

Reply:
hmm, I'd push back on this. we set this as a default behavior at a higher level of abstraction than the storage layer getter method, and I'd prefer to keep it there. I'd like the storage layer to be consistent, i.e. if `get_matching_pool_for_order` says that order `o` is in pool `p`, it's iff there is a table entry mapping `o -> p`.